src/test/java/com/sigma/cinematicketpro/service/impl/MovieServiceImplTest.java

```diff
@@ -0,0 +1,138 @@
+package com.sigma.cinematicketpro.service.impl;
+
+import com.sigma.cinematicketpro.TestUtils;
+import com.sigma.cinematicketpro.entity.Movie;
+import com.sigma.cinematicketpro.exception.ResourceNotFoundException;
+import com.sigma.cinematicketpro.repository.MovieRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class MovieServiceImplTest {
+
+    @Mock
+    private MovieRepository movieRepository;
+    @InjectMocks
+    private MovieServiceImpl sut;
+
+    private static final List<Movie> movieList = TestUtils.getValidMovies();
+
+    @Test
+    void shouldReturnListOfMoviesFromRepository_WhenGetAllMovies() {
+        List<Movie> expectedMovieList = movieList;
+        when(movieRepository.findAll()).thenReturn(expectedMovieList);
+
+        List<Movie> actualMovieList = sut.getAllMovies();
+
+        assertThat(expectedMovieList).isEqualTo(actualMovieList);
```

Comment:
this is fun ;) yep assert that mock works properly ;)

Reply:
not exactly like this) I wanted to make sure that no other changes have been made to the list of films returned by the repo. Overall, l I agree with you, so I changed the test to check the invocation of a findAll method in a MovieRepository.